backend/ctgov/api/views.py

```diff
@@ -676,3 +743,514 @@ def filter_phase(request):
+            "primary_completion_date",
+            primary_completion_date_from,
+            primary_completion_date_to,
+        )
+        + get_date_range(
+            "study_first_posted_date",
+            first_posted_date_from,
+            first_posted_date_to,
+        )
+        + get_date_range(
+            "results_first_posted_date",
+            results_first_posted_date_from,
+            results_first_posted_date_to,
+        )
+        + get_date_range(
+            "last_update_posted_date",
+            last_update_posted_date_from,
+            last_update_posted_date_to,
+        )
+        + get_exists_query(
+            "results_first_posted_date", study_results_query_type
+        )
+        + get_match_query("study_type", study_type)
+        + ", "
+        + get_match_query("eligibility_gender", eligibility_gender)
+        + ", "
+        + get_match_query("eligibility_criteria", eligibility_ethnicity)
+        + ", "
+        + get_match_query("eligibility_criteria", eligibility_condition)
+        + ", "
+        + get_match_query("sponsor_name", study_collaborator)
+        + ", "
+        + get_match_query("study_ids", study_ids)
+        + ", "
+        + get_match_query("location_name", study_location_terms)
+        + ", "
+        + get_match_query("funder_type", study_funder_type)
+        + ", "
+        + get_exists_query(
+            "results_submitted_qc_not_done", query_type_qc_not_done
+        )
+        + get_exists_query("results_submitted_qc_done", query_type_qc_done)
+        + get_match_query("document_types", study_document_type)
+        + ", "
+        + get_match_query("official_title", study_title_acronym)
+        + ", "
+        + get_match_query("acronym", study_title_acronym)
+        + ", "
+        + get_match_query("primary_outcome_measures", study_outcome_measure)
+        + ", "
+        + get_match_query("secondary_outcome_measures", study_outcome_measure)
+        + ", "
+        + get_gte_query(
+            "eligibility_min_age_numeric", eligibility_age_lower_limit
+        )
+        + ", "
+        + get_lte_query(
+            "eligibility_max_age_numeric", eligibility_age_upper_limit
+        )
+        + ", "
+        + get_range_query(
+            "eligibility_min_age_numeric",
+            eligibility_age_group_lower_limit_1,
+            eligibility_age_group_upper_limit_1,
+        )
+        + ", "
+        + get_range_query(
+            "eligibility_max_age_numeric",
+            eligibility_age_group_lower_limit_2,
+            eligibility_age_group_upper_limit_2,
+        )
+        + ", "
+        + get_match_query("study_phase", phase)
+        + ", "
+        + get_match_query("study_brief_desc", study_roa)
+        + "] "
+        + "} "
+        + "} "
+        + "} "
+    )
+
+    return filters
+
+
+# Function to construct fuzzy match query json for elastic search
+def get_match_query(field, param):
+    match_query = (
+        "{ "
```

Comment:
Same comment as above. Use a dictionary then use json.dumps to convert it to a string.